src/Highlighter.ts

```diff
@@ -162,6 +174,22 @@ export default class Highlighter {
     this.onClick(event);
   }
 
+  private onFocusHandler = (type: 'in' | 'out') => (ev: Event): void => {
+    const handler = type === 'in' ? this.options.onFocusIn : this.options.onFocusOut;
+    if (!handler) {
+      return;
+    }
+
+    const highlightId = isHtmlElement(ev.target) && ev.target.hasAttribute(DATA_SCREEN_READERS_ATTR)
+      ? ev.target.getAttribute(DATA_ID_ATTR)
+      : null;
+    const highlight = highlightId ? this.getHighlight(highlightId) : null;
+
+    if (highlight) {
+      handler(highlight);
```

Comment:
in order for the highlight to actually get the "focus styles", are we relying on the implementation to call `highlight.addFocusedStyles` inside this handler?

is there a reason not to call that on focusin by default?

Reply:
currently in `highlightManager.ts` we are calling `highlighter.clearFocus()` and then we call `...addFocusedStyles` on every `update`
This is required to make sure that proper highlight is focused when user focuses it by clicking on the card
So we could add here:
```
this.clearFocusedStyles();
highlight.addFocusedStyles();
```
however we still have to do the same in `highlightManager`

Reply:
it'd be nice if the click to focus worked without any userland implementation because rex isn't the only user of this library

Reply:
this is still outstanding right? if you're concerned with duplicate function calls you could do it as a default implementation of the `onFocusIn` and `onFocusOut` options